docs/components/badge.md

```diff
@@ -1,276 +1,321 @@
+  <KBadge appearance="warning">
+    <template #icon>
+      <WarningOutlineIcon />
+    </template>
+    4
+  </KBadge>
+  <KBadge :icon-before="false">
+    Badge with icon
+    <template #icon>
+      <KongIcon />
+    </template>
+  </KBadge>
 </div>
 
 ```html
-<KBadge truncation-tooltip="Truncation unnecessary">Truncation unnecessary</KBadge>
-<KBadge truncation-tooltip="Hey! Let me see that awesome truncation">Hey! Let me see that awesome truncation</KBadge>
-```
-
-### forceTooltip
-
-If you want to show the tooltip regardless of whether the badge text is truncated, use this prop.
-
-<KBadge truncation-tooltip="But I'm tooltipping anyways" force-tooltip>Truncation unnecessary</KBadge>
-
-```html
-<KBadge truncation-tooltip="But I'm tooltipping anyways" force-tooltip>Truncation unnecessary</KBadge>
+<KBadge appearance="warning">
+  <template #icon>
+    <WarningOutlineIcon />
+  </template>
+  4
+</KBadge>
+<KBadge :icon-before="false">
+  Badge with icon
+  <template #icon>
+    <KongIcon />
+  </template>
+</KBadge>
 ```
 
-### maxWidth
+:::tip TIP
+If you want to make an icon clickable, you can assign `role="button"` and appropriate `tabindex` attributes to that element and bind an event handler. KBadge will take care of state styling (hover, focus, disabled).
+
+
+<Transition name="kongponents-fade-transition" mode="out-in">
+  <KBadge
+    appearance="warning"
+    :icon-before="false"
+    v-if="showDismissableBadge"
+    key="badge"
+  >
+    Dismiss me
+    <template #icon>
+      <CloseIcon
+        role="button"
+        tabindex="0"
+        @click="handleIconClick"
+      />
+    </template>
+  </KBadge>
+  <KButton v-else key="reset-button" size="small" @click="showDismissableBadge = true">Reset</KButton>
+</Transition>
+
+```vue
+<template>
+  <KBadge
+    appearance="warning"
+    :icon-before="false"
+    v-if="showBadge"
+  >
+    Dismiss me
+    <template #icon>
+      <CloseIcon
+        role="button"
+        tabindex="0"
+        @click="handleIconClick"
+      />
+    </template>
+  </KBadge>
+</template>
 
-Max width to apply truncation at. Works just like `width` property in CSS. Default value is `200px`. Is superseded by `--KBadgeMaxWidth` [CSS variable](#theming) if both present.
-
-<KBadge max-width="50px" truncation-tooltip="Tooltip often">Truncate early</KBadge>
-
-```html
-<KBadge max-width="50px" truncation-tooltip="Tooltip often">Truncate early</KBadge>
-```
-
-## Slots
+<script setup lang="ts">
+import { ref } from 'vue'
 
-- `default` - innerHTML inside badge
+const showBadge = ref<boolean>(true)
 
-<KBadge appearance="success">SUCCESS</KBadge>
+const handleIconClick = () => {
+  showBadge.value = !showBadge.value
+}
+</script>
+:::
 
-```html
-<KBadge appearance="success">SUCCESS</KBadge>
-```
+KBadge takes care of icon color, size and spacing as long as you use icons sourced from [@kong/icons](https://github.com/Kong/icons) package.
```

Comment:
suggestion: can you move this sentence above the TIP block?

Reply:
It kinda logically ties to the TIP block below.